samples/nrf9160/modem_shell/src/location/location_shell.c

```diff
@@ -0,0 +1,413 @@
+	} else if (strcmp(argv[1], "cancel") == 0) {
+		loc_cmd_args.command = LOCATION_CMD_CANCEL;
+	} else {
+		shell_error(shell, "Unsupported command=%s\n", argv[1]);
+		ret = -EINVAL;
+		goto show_usage;
+	}
+
+	/* Reset getopt due to possible previous failures */
+	optreset = 1;
+
+	/* We start from subcmd arguments */
+	optind = 2;
+
+	while ((opt = getopt_long(argc, argv, "", long_options, &long_index)) != -1) {
+		switch (opt) {
+		case LOCATION_SHELL_OPT_GNSS_TIMEOUT:
+			gnss_timeout = atoi(optarg);
+			if (gnss_timeout == 0) {
+				shell_error(shell, "GNSS timeout (%d) must be positive integer. ",
+					    gnss_timeout);
+				return -EINVAL;
+			}
+			gnss_timeout_set = true;
+			break;
+
+		case LOCATION_SHELL_OPT_CELLULAR_TIMEOUT:
+			cellular_timeout = atoi(optarg);
+			if (cellular_timeout == 0) {
+				shell_error(shell,
+					    "Cellular timeout (%d) must be positive integer. ",
+					    cellular_timeout);
+				return -EINVAL;
+			}
+			cellular_timeout_set = true;
+			break;
+
+		case LOCATION_SHELL_OPT_INTERVAL:
+			interval = atoi(optarg);
+			gnss_interval_set = true;
+			break;
+
+		case LOCATION_SHELL_OPT_GNSS_ACCURACY:
+			if (strcmp(optarg, "low") == 0) {
+				gnss_accuracy = LOC_ACCURACY_LOW;
+			} else if (strcmp(optarg, "normal") == 0) {
+				gnss_accuracy = LOC_ACCURACY_NORMAL;
+			} else if (strcmp(optarg, "high") == 0) {
+				gnss_accuracy = LOC_ACCURACY_HIGH;
+			} else {
+				shell_error(shell, "Unknown GNSS accuracy. See usage:");
+				goto show_usage;
+			}
+			gnss_accuracy_set = true;
+			break;
+		case LOCATION_SHELL_OPT_GNSS_NUM_FIXES:
+			gnss_num_fixes = atoi(optarg);
+			gnss_num_fixes_set = true;
+			break;
+		case LOCATION_SHELL_OPT_METHOD_1:
+			if (strcmp(optarg, "cellular") == 0) {
+				method1 = LOC_METHOD_CELLULAR;
+			} else if (strcmp(optarg, "gnss") == 0) {
+				method1 = LOC_METHOD_GNSS;
+			} else {
+				shell_error(shell, "Unknown method given. See usage:");
+				goto show_usage;
+			}
+			method_count++;
+			method1_set = true;
+			break;
+		case LOCATION_SHELL_OPT_METHOD_2:
+			if (strcmp(optarg, "cellular") == 0) {
+				method2 = LOC_METHOD_CELLULAR;
+			} else if (strcmp(optarg, "gnss") == 0) {
+				method2 = LOC_METHOD_GNSS;
+			} else {
+				shell_error(shell, "Unknown method given. See usage:");
+				goto show_usage;
+			}
+			method_count++;
+			method2_set = true;
+			break;
+		case '?':
+			goto show_usage;
+			break;
+		default:
+			shell_error(shell, "Unknown option. See usage:");
+			goto show_usage;
+		}
+	}
+
+	/* Handle location subcommands */
+	switch (loc_cmd_args.command) {
+	case LOCATION_CMD_CANCEL:
+		ret = location_request_cancel();
+		if (ret) {
+			shell_error(shell, "Cannot cancel location request, err: %d", ret);
+			return -1;
+		}
+		shell_print(shell, "Getting of location cancelled");
+		break;
+
+	case LOCATION_CMD_GET: {
+		struct loc_config config = { 0 };
+		struct loc_method_config methods[LOC_MAX_METHODS] = { 0 };
+		struct loc_config *real_config = &config;
```

Comment:
Use location api default setting functions here and override values based on what variables are given
@trantanen

Reply:
Done